src/evmasm.js

```diff
@@ -154,7 +155,11 @@ class Contract {
     }
 
     push(data) {
-        const d = Web3Utils.hexToBytes(Web3Utils.toHex(data));
+        // The code resembles previously used Web3Utils.hexToBytes from web3@1.0.0-beta.55
+        const d = (typeof data === 'string') ? Array.from(hexToBytes(data.replace(/^0x/, '')))
+            : (data === -1) ? [-0, NaN] // broken; but worked like this before
+            : (data instanceof Uint8Array) ? Array.from(data)
+            : Array.from(numberToVarBytesBE(data));
```

Comment:
this is quite hard to read as it has nested trinaries

Reply:
After much thought I have decided to leave this as-is.  I wish JavaScript had if/else blocks as expressions, but since it doesn't we are left with either a single-use poorly named function, an immediately called anonymous function, a `let` instead of a `const`, or a ternary.  While I can appreciate that the ternary is hard to read for those not used to using them in this way, I think it is better than the other options on the table.